gneiss/util.py

```diff
@@ -0,0 +1,172 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, gneiss development team.
+#
+# Distributed under the terms of the GPLv3 License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+import warnings
+
+
+def match(table, metadata, intersect=False):
+    """ Sorts samples in metadata and contingency table in the same order.
+
+    Parameters
+    ----------
+    table : pd.DataFrame
+        Contingency table where samples correspond to rows and
+        features correspond to columns.
+    metadata: pd.DataFrame
+        Metadata table where samples correspond to rows and
+        explanatory metadata variables correspond to columns.
+    intersect : bool, optional
+        Specifies if only the intersection of samples in the
+        contingency table and the metadata table will returned.
+        By default, this is False.
+
+    Returns
+    -------
+    pd.DataFrame :
+        Filtered contingency table.
+    pd.DataFrame :
+        Filtered metadata table
+
+    Raises
+    ------
+    ValueError:
+        Raised if duplicate sample ids are present in `table`.
+    ValueError:
+        Raised if duplicate sample ids are present in `metadata`.
+    ValueError:
+        Raised if `table` and `metadata` have incompatible sizes.
+    """
+    subtableids = set(table.index)
+    submetadataids = set(metadata.index)
+    if len(subtableids) != len(table.index):
+        raise ValueError("`table` has duplicate sample ids.")
+    if len(submetadataids) != len(metadata.index):
+        raise ValueError("`metadata` has duplicate sample ids.")
+
+    if intersect:
+        idx = subtableids & submetadataids
+        idx = sorted(idx)
```

Comment:
Because otherwise the pandas dataframe will be scrambled due to the way that sets work.

Sorting them the only way to resolve this issue, for the sake of testing.